litigious-liberators/profile_cards/profile_cards.py

```diff
@@ -0,0 +1,210 @@
+        )
+
+    def on_enter(self, *args):
+        if self.sound:
+            self.sound.play()
+        return super().on_enter(*args)
+
+
+class ProfileList(ScreenManager):
+    _attributes = DictProperty(None)
+
+    def __init__(self, **kwargs):
+        super(ProfileList, self).__init__(**kwargs)
+        self.profile_dir = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "../profiles/write-ups"
+        )
+        help = Label(
+            text="""Your goal is to maximize all three attributes: Knowledge, Welfare,Energy:
+        Use left/right arrowkeys to swipe the cards: each decision will influence your attributes!
+        Press on each card image if you want to learn more about it ;)""",
+            text_size=(self.width * 3, None),
+        )
+        self.help = Popup(title="Help", content=help, size_hint=(0.5, 0.7))
+        # self.help.content.button.bind(on_press=self.help.dismiss)
+        self.static_profile_list = listdir(self.profile_dir)
+        self.profile_list = deepcopy(self.static_profile_list)
+        self.cycler = self.r_cycle(self.profile_list)
+        #  should be initialisable in main app
+        self._attributes = Counter({"Knowledge": 5, "Welfare": 5, "Energy": 5})
+        # self._keyboard = Window.request_keyboard(self._keyboard_closed, self)
+        # self._keyboard.bind(on_key_down=self._on_keyboard_down)
+        with open(f"{self.profile_dir}/{next(self.cycler)}", "r") as profile_file:
+            profile = safe_load(profile_file.read())
+            self.add_widget(ProfileCard(profile))
+
+    @property
+    def attributes(self):
+        return self._attributes
+
+    @attributes.setter
+    def attributes(self, value):
+        self._attributes = value
+        for k, v in self._attributes.items():
+            self._attributes[k] = min(10, v)
+        if any(x < 0 for x in self._attributes.values()):
+            App.get_running_app().root.current = "loss_screen"
+        elif all(x == 10 for x in self._attributes.values()):
+            App.get_running_app().root.current = "win_screen"
+        print(self.attributes)
+
+    @staticmethod
+    def r_cycle(x):
+        while True:
+            shuffle(x)
+            for element in x:
+                yield element
+
+    def card_swipe(self, direction):
+        next_profile = next(self.cycler)
+        with open(f"{self.profile_dir}/{next_profile}", "r") as profile_file:
+            profile = safe_load(profile_file.read())
+        current = self.current_screen
+        trans = SlideTransition()
+        totals = Counter()
+        totals.update(self.attributes)
+        next_card = ProfileCard(profile)
+        delta = current.deltas[direction]
+        totals.update(delta)
+        self.attributes = totals
+        message = direction.upper() + "\n" + current.results[direction]
+        for k, v in delta.items():
+            message += f"{k}: {v if v < 0 else '+' + str(v)}\n"
```

Comment:
k and v are not descriptive enough variable names, what are k and v? Variables should not be one character.